[PLT-XXX] Improve tag writing

joshuafleck opened the pull request:
This improves tag writing such that written tags can be re-parsed with the existing tag parsing logic. There were two cases where we were able to write tags that could not be parsed:
1. When the tag had a single colon
2. When the tag contained a space

This will reduce the amount of unparseable metrics we're writing, which will reduce the amount of metrics we're currently losing due to parsing failures.
<img width="870" alt="Screenshot 2024-02-05 at 15 24 43" src="https://github.com/meetcleo/statsd-instrument/assets/1008898/3d0adfff-85cb-48b9-90f0-6767c397bfae">

From: https://cleo.app.coralogix.us/#/dashboards/MzXT3l2IP4torRe531Nck

joshuafleck committed:
[PLT-XXX] Improve tag writing

This improves tag writing such that written tags can be re-parsed with the existing tag parsing logic. There were two cases where we were able to write tags that could not be parsed:
1. When the tag had a single colon
2. When the tag contained a space

This will reduce the amount of unparseable metrics we're writing, which will reduce the amount of metrics we're currently losing due to parsing failures.

flood4life commented on lib/statsd/instrument/datagram_builder.rb:
```diff
@@ -70,20 +72,20 @@ def latency_metric_type
         # Fast path when no string replacement is needed
-        return tags unless tags.any? { |tag| /[|,]/.match?(tag) }
+        return tags unless tags.any? { |tag| /[|, ]/.match?(tag) }
 
-        tags.map { |tag| tag.tr("|,", "") }
+        tags.map { |tag| tag.tr("|, ", REPLACEMENT_CHAR) }
```
Are we deliberately using the single whitespace literal here rather than `\s` that would catch a wider variety of whitespaces?

joshuafleck replied:
Good shout, updated in 196b0cc

joshuafleck committed:
[PLT-XXX] Make tag normalization look for any space char

Previously, this just checked for single space chars, but it's safer to remove any type of whitespace if we find it.
